commandsv3/src/main/java/org/wpilib/commands3/Scheduler.java

```diff
@@ -0,0 +1,891 @@
+    for (var state : m_commandStates.values()) {
+      if (ancestors.contains(state)) {
+        // Can't conflict with an ancestor command
+        continue;
+      }
+
+      var c = state.command();
+      if (c.conflictsWith(command) && command.isLowerPriorityThan(c)) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  private void evictConflictingOnDeckCommands(Command command) {
+    for (var iterator = m_onDeck.iterator(); iterator.hasNext(); ) {
+      var scheduledState = iterator.next();
+      var scheduledCommand = scheduledState.command();
+      if (scheduledCommand.conflictsWith(command)) {
+        // Remove the lower priority conflicting command from the on deck commands.
+        // We don't need to call removeOrphanedChildren here because it hasn't started yet,
+        // meaning it hasn't had a chance to schedule any children
+        iterator.remove();
+        emitEvent(interrupted(scheduledCommand, command));
+      }
+    }
+  }
+
+  /**
+   * Cancels all running commands with which an incoming state conflicts. Ancestor commands of the
+   * incoming state will not be canceled.
+   */
+  @SuppressWarnings("PMD.CompareObjectsWithEquals")
+  private void evictConflictingRunningCommands(CommandState incomingState) {
+    // The set of root states with which the incoming state conflicts but does not inherit from
+    Set<CommandState> conflictingRootStates =
+        m_commandStates.values().stream()
+            .filter(state -> incomingState.command().conflictsWith(state.command()))
+            .filter(state -> !inheritsFrom(incomingState, state.command()))
+            .map(
+                state -> {
+                  // Find the highest level ancestor of the conflicting command from which the
+                  // incoming state does _not_ inherit. If they're totally unrelated, this will
+                  // get the very root ancestor; otherwise, it'll return a direct sibling of the
+                  // incoming command
+                  CommandState root = state;
+                  while (root.parent() != null && root.parent() != incomingState.parent()) {
+                    root = m_commandStates.get(root.parent());
+                  }
+                  return root;
+                })
+            .collect(Collectors.toSet());
+
+    // Cancel the root commands
+    for (var conflictingState : conflictingRootStates) {
+      emitEvent(interrupted(conflictingState.command(), incomingState.command()));
+      cancel(conflictingState.command());
+    }
+  }
+
+  /**
+   * Checks if a particular command is an ancestor of another.
+   *
+   * @param state the state to check
+   * @param ancestor the potential ancestor for which to search
+   * @return true if {@code ancestor} is the direct parent or indirect ancestor, false if not
+   */
+  @SuppressWarnings({"PMD.CompareObjectsWithEquals", "PMD.SimplifyBooleanReturns"})
+  private boolean inheritsFrom(CommandState state, Command ancestor) {
```

Comment:
"inherits" isn't the right word here

perhaps `isAncestorOf` would be better (note it flips operand order)